apply_DDT.py

```diff
@@ -0,0 +1,569 @@
+#==============================================================================
+# apply_DDT.py ----------------------------------------------------------------
+#------------------------------------------------------------------------------
+# Author(s): Brendan Regnery, Sara Nabili -------------------------------------
+#------------------------------------------------------------------------------
+# Applies a DDT to a trained BDT model ----------------------------------------
+#    (Designed Decorrelated Tagger, https://arxiv.org/pdf/1603.00027.pdf) -----
+#------------------------------------------------------------------------------
+
+import os, os.path as osp, glob, pickle, logging, warnings, json, math, re
+from time import strftime
+from collections import OrderedDict
+from common import create_DDT_map_dict, calculate_varDDT
+import json
+import argparse
+
+import numpy as np
+import matplotlib
+matplotlib.use('Agg') # prevents opening displays (fast), must use before pyplot
+import matplotlib.pyplot as plt
+from sklearn.metrics import confusion_matrix, roc_curve, roc_auc_score, auc
+import pandas as pd
+import xgboost as xgb
+from scipy.ndimage import gaussian_filter
+import mplhep as hep
+hep.style.use("CMS") # CMS plot style
+
+#disable tex formating while having problems
+#plt.rcParams['text.usetex'] = False
+
+np.random.seed(1001)
+
+from common import logger, DATADIR, Columns, time_and_log, imgcat, set_mpl_fontsize, columns_to_numpy
+
+#------------------------------------------------------------------------------
+# Global variables and user input arguments -----------------------------------
+#------------------------------------------------------------------------------
+
+training_features = [
+    'girth', 'ptd', 'axismajor', 'axisminor',
+    'ecfm2b1', 'ecfd2b1', 'ecfc2b1', 'ecfn2b2', 'metdphi',
+    'ak15_chad_ef', 'ak15_nhad_ef', 'ak15_elect_ef', 'ak15_muon_ef', 'ak15_photon_ef', 
+    ]
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description='Process some inputs.')
+
+    parser.add_argument('--bkg_data_files', default='data/bkg_20240515/Summer20UL18/*.npz', help='Background data files')
+    parser.add_argument('--sig_data_files', default='data/signal/*mDark-10_rinv-0p3*.npz', help='Signal data files')
```

Comment:
let's reserve the term "data" for actual observed data. just call these `--bkg-files` and `--sig-files`
(dashes are preferred in command-line arguments rather than underscores because they're easier to type; for the python dest variable, argparse automatically converts them to underscores)